library/bignum.c

```diff
@@ -1720,6 +1728,20 @@ int mbedtls_mpi_exp_mod(mbedtls_mpi *X, const mbedtls_mpi *A,
     return ret;
 }
 
+int mbedtls_mpi_exp_mod(mbedtls_mpi *X, const mbedtls_mpi *A,
+                        const mbedtls_mpi *E, const mbedtls_mpi *N,
+                        mbedtls_mpi *prec_RR)
+{
+    return mbedtls_mpi_exp_mod_optionally_safe(X, A, E, MBEDTLS_MPI_IS_SECRET, N, prec_RR);
+}
+
+int mbedtls_mpi_exp_mod_unsafe(mbedtls_mpi *X, const mbedtls_mpi *A,
```

Comment:
... and this is causing FreeBSD CI builds to fail

```
[2024-08-22T12:59:17.807Z] /usr/home/ec2-user/workspace/mbed-tls-pr-head_PR-9281-head/src/library/bignum.c:1738:5: error: no previous prototype for function 'mbedtls_mpi_exp_mod_unsafe' [-Werror,-Wmissing-prototypes]
[2024-08-22T12:59:17.807Z] int mbedtls_mpi_exp_mod_unsafe(mbedtls_mpi *X, const mbedtls_mpi *A,
[2024-08-22T12:59:17.807Z]     ^
[2024-08-22T12:59:17.807Z] /usr/home/ec2-user/workspace/mbed-tls-pr-head_PR-9281-head/src/library/bignum.c:1738:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
[2024-08-22T12:59:17.807Z] int mbedtls_mpi_exp_mod_unsafe(mbedtls_mpi *X, const mbedtls_mpi *A,
[2024-08-22T12:59:17.807Z] ^
[2024-08-22T12:59:17.807Z] static 
[2024-08-22T12:59:17.807Z] [  8%] Building C object library/CMakeFiles/mbedcrypto.dir/bignum_core.c.o
[2024-08-22T12:59:17.807Z] 1 error generated.
[2024-08-22T12:59:17.807Z] make[2]: *** [library/CMakeFiles/mbedcrypto.dir/build.make:268: library/CMakeFiles/mbedcrypto.dir/bignum.c.o] Error 1
[2024-08-22T12:59:17.807Z] make[2]: *** Waiting for unfinished jobs....
[2024-08-22T12:59:18.064Z] make[1]: *** [CMakeFiles/Makefile2:1053: library/CMakeFiles/mbedcrypto.dir/all] Error 2
[2024-08-22T12:59:18.064Z] make: *** [Makefile:146: all] Error 2
[2024-08-22T12:59:18.064Z] ^^^^test_cmake_out_of_source: build: cmake 'out-of-source' build: make -> 2^^^^
```

Reply:
Yes, we have `-Wmissing-prototypes` in some builds since very recently.

We have a precedent of creating a header in `library` in an LTS branch: `bignum_internal` was added 2.28.7. So please (re)create `library/bignum_internal.h` for `mbedtls_mpi_exp_mod_unsafe()`.